### PR TITLE
Changed DataFilter to automatically add search to SelectMultiple when more than 10 options

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed DataFilter to automatically add search to SelectMultiple when more than 12 options.

#### Where should the reviewer start?

DataFilter.js

#### What testing has been done on this PR?

Manually.
Added a unit test to verify that the search input is added.

#### How should this be manually tested?

manually

#### Do Jest tests follow these best practices?

followed existing test patterns

#### Do the grommet docs need to be updated?

yes

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/DataFilter/DataFilter.js [hunk 1]
```diff
@@ -8,6 +8,10 @@ import { RangeSelector } from '../RangeSelector';
 import { SelectMultiple } from '../SelectMultiple';
 import { DataFilterPropTypes } from './propTypes';
 
+// empirical constants for when we change inputs
+const maxCheckBoxGroupOptions = 7;
+const minSelectSearchOptions = 12;
```
Comment (halocline) [inline src/js/components/DataFilter/DataFilter.js:13]: When researching SelectMultiple, the team set these thresholds at:
 - CheckBoxGroup = 1-4
 - SelectMultiple = 5-9
 - SelectMultiple with search = 10+
 
Do we want to align with guidance on the HPE Design System ([min](https://design-system.hpe.design/components/selectmultiple#when-not-to-use),[max](https://design-system.hpe.design/components/selectmultiple#best-practices))?
```suggestion
// empirical constants for when we change inputs
const maxCheckBoxGroupOptions = 4;
const minSelectSearchOptions = 10;
```

File: src/js/components/DataFilter/DataFilter.js [hunk 2]
```diff
@@ -108,15 +127,21 @@ export const DataFilter = ({
...
       } else {
         content = (
           <SelectMultiple
             id={id}
             name={property}
             showSelectedInline
-            options={options}
+            options={searchedOptions}
+            onSearch={
+              options.length > minSelectSearchOptions
```
Comment (halocline) [inline src/js/components/DataFilter/DataFilter.js:140]: If defined as "min" check should include the min, correct?

```suggestion
              options.length >= minSelectSearchOptions
```

**Review by halocline — COMMENTED** [review]
Functionality and tests look good. Comments related to polishing details.

**Review by halocline — APPROVED** [review]

**Comment by halocline** [comment]
> The work here looks good -- I'm not sure why the background chromatic snapshot is changing?

It looks like those two examples in the snapshot use the HPE theme. Could the diff be due to referencing new/different Metric font files from the theme?

**Comment by jcfilben** [comment]
> > The work here looks good -- I'm not sure why the background chromatic snapshot is changing?
>
> It looks like those two examples in the snapshot use the HPE theme. Could the diff be due to referencing new/different Metric font files from the theme?

Yes I believe this is coming from the new font files in the hpe theme. This change came in previously I just didn't update the snapshot on chromatic. It should be fixed to the correct snapshot now.

**Review by jcfilben — APPROVED** [review]
Looks good!

**Pull request merged**